### ISSUE TITLE
The requested check was never run against this ref, exiting...

### ISSUE THREAD

**Issue by ayozemr** [lewagon/wait-on-check-action#7]
Hello!

I am using your action to solve a problem I have:
- Workflow builds api on every push
- We want e2e workflow to run on PRs to master -> Runs using build artifact

So I set up a new workflow with your action, I used github.sha as ref because it runs in pushes to the PR.

on:
  pull_request:
    types: [opened, synchronize, reopened, ready_for_review]
    branches:
      - master

name: Wait until build is complete
        uses: lewagon/wait-on-check-action@v0.1
        with:
          ref: ${{ github.sha }} # can be commit SHA or tag too
          check-name: build # name of the existing check - omit to wait for all checks
          repo-token: ${{ secrets.GITHUB_TOKEN }}
          wait-interval: 15 # seconds

I got this error. Maybe they triggered at the same time and e2e run before build job was started? I dont know if the correct way to implement it is to place some "sleep 15" before your wait on check 🤔 
`The requested check was never run against this ref, exiting...`

Thank you for your time!

**Comment by YonatanKra** [comment]
I get the same thing....

**Comment by rrijkse** [comment]
I solved this by running with `uses: lewagon/wait-on-check-action@master` haven't done a deep dive to figure out what's changed since the release but in my case it I was trying to use the feature from PR #6 which doesn't seem to be released yet.

**Comment by ayozemr** [comment]
> I solved this by running with `uses: lewagon/wait-on-check-action@master` haven't done a deep dive to figure out what's changed since the release but in my case it I was trying to use the feature from PR #6 which doesn't seem to be released yet.

Thanks @rrijkse , will try

**Comment by progapandist** [comment]
Let me know if it worked! I think I have to get serious about it and start a Changelog and regular release shedule, will be on it right after the New Year! For now using `master` is the best bet

**Comment by owen2345** [comment]
Hey here, I am having the same problem.
I have 3 testing workflows + 1 continuous deployment workflow.
After the first 3 workflows successfully completed, CD workflow "CD" is still stuck:
<img width="452" alt="Screen Shot 2021-01-07 at 6 06 13 AM" src="https://user-images.githubusercontent.com/1148415/103879910-d3a0e280-50ae-11eb-8a03-11aa676388d6.png">

**Hints**:
Maybe the checker is expecting for empty pending workflows but my CD workflow is still in progress, then it will never be processed.

**Comment by progapandist** [comment]
hi @owen2345 — can you please send me the example code from your workflow that uses the `wait-on-check`?

**Comment by owen2345** [comment]
Sure @progapandist 
```
name: Continous deployment

on:
  push:
    branches:
      - master
      - staging

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

jobs:
  deploy:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      # This step will retry until required check passes
      # and will fail the whole workflow if the check conclusion is not a success
      - name: Wait on tests
        uses: lewagon/wait-on-check-action@master
        with:
          ref: ${{ github.event.pull_request.head.sha || github.sha }}
          # check-name: test # name of the existing check - omit to wait for all checks
          repo-token: ${{ secrets.GITHUB_TOKEN }}
          wait-interval: 30 # seconds

      - name: Continuos deployment
        run: "..."

```
Please let me know if you need anything else to find the problem.
Thank you!

**Comment by progapandist** [comment]
Thank you! Try passing `ref: ${{ github.sha }}` instead of `ref: ${{ github.event.pull_request.head.sha || github.sha }}`. Frankly, I don't understand the use case for having a conditional between SHAs

**Comment by progapandist** [comment]
Maybe omitting check name for all checks doesn't work as expected. Can you try waiting for all three testing checks explicitly one after another and see if that works?

**Comment by owen2345** [comment]
Unfortunately it did not work (the same problem "it still stuck").
Additionally I noticed that when one of the workflows failed, my CD workflow is still stuck. I think it should fail/cancel too.

**Comment by progapandist** [comment]
Unfortunately I don’t see an immediate answer. Try referring to wait-on-check-action@v0.1, do one test at a time, and pass check names explicitly. If it works, then it must be a regression in master but I have no time to investigate urgently.

**Comment by matiasalbarello** [comment]
The problem seems to be that the workflow is waiting for itself to finish in order to continue. When the `check-name` is specified, this is not a problem.

**Comment by progapandist** [comment]
Yes, that means https://github.com/lewagon/wait-on-check-action/pull/6 probably messed it up. Sorry for the oversight. I will try to investigate, but haven’t got a lot of free time on my hands rn. If someone can see the problem clearly and make a quick PR - that will be much appreciated!

**Comment by progapandist** [comment]
For now the solution would be to wait on check name explicitly, don’t omit it, as @matiasalbarello mentioned (thanks!)

**Comment by progapandist** [comment]
Should be fixed thanks to https://github.com/lewagon/wait-on-check-action/pull/10 (🙌  @matiasalbarello), please check and confirm so I can close this issue

**Comment by codetheweb** [comment]
I believe #10 introduced a new issue:
https://github.com/Baton-donation/backend/runs/1682419913?check_suite_focus=true#step:5:7

**Comment by progapandist** [comment]
Thanks for letting me know, will fix tonight

**Comment by progapandist** [comment]
https://github.com/lewagon/wait-on-check-action/commit/bafe56a6863672c681c3cf671f5e10b20abf2eaa — this should fix it, please check @codetheweb 🙏

**Comment by codetheweb** [comment]
Looks like it's working now, thank you!

Could you tag it too please so I can pin the version?

**Comment by progapandist** [comment]
Sure, will do it now, it's a long time coming...

**Comment by progapandist** [comment]
https://github.com/lewagon/wait-on-check-action/releases/tag/v0.2 🎉  big thanks to everyone involved! I will close the issue, feel free to reopen if any more corner cases surface
